`openjdk17` FTBFS on WSL2 specifically

```
       > checking for greadlink... [not found]
       > checking for readlink... /nix/store/a7gvj343m05j2s32xcnwr35v31ynlypr-coreutils-9.1/bin/readlink
       > checking for cygpath... [not found]
       > checking for wslpath... [not found]
       > checking for lsb_release... [not found]
       > checking for cmd.exe... [not found]
       > checking for cmp... /nix/store/j9p3g8472iijd50vhdprx0nmk2fqn5gv-diffutils-3.8/bin/cmp
       > checking for uniq... /nix/store/a7gvj343m05j2s32xcnwr35v31ynlypr-coreutils-9.1/bin/uniq
       > checking build system type... x86_64-pc-wsl
       > checking host system type... x86_64-pc-wsl
       > checking target system type... x86_64-pc-wsl
       > checking openjdk-build os-cpu... windows-x86_64
       > checking openjdk-target os-cpu... windows-x86_64
       > checking compilation type... native
       > checking Windows environment type... wsl1
       > configure: error: Incorrect wsl1 installation. Neither cygpath nor wslpath was found
       > configure exiting with result code 1
```

They're trying to be smart: https://github.com/openjdk/jdk17u-dev/blob/da6fca4d78465625775f7b384ba752185694f699/make/autoconf/build-aux/config.guess#L59

We should probably patch that check out, but maybe people with actual Java experience will have better ideas.

@edwtjo @asbachb

--- a/pkgs/development/compilers/openjdk/13.nix
+++ b/pkgs/development/compilers/openjdk/13.nix
@@ -61,6 +61,11 @@ let
       patchShebangs --build configure
     '';
 
+    # JDK's build system attempts to specifically detect
+    # and special-case WSL, and we don't want it to do that,
+    # so pass the correct platform names explicitly
+    configurePlatforms = ["build" "host"];
+
     configureFlags = [
       "--with-boot-jdk=${openjdk-bootstrap.home}"
       "--with-version-pre="

--- a/pkgs/development/compilers/openjdk/14.nix
+++ b/pkgs/development/compilers/openjdk/14.nix
@@ -56,6 +56,11 @@ let
       patchShebangs --build configure
     '';
 
+    # JDK's build system attempts to specifically detect
+    # and special-case WSL, and we don't want it to do that,
+    # so pass the correct platform names explicitly
+    configurePlatforms = ["build" "host"];
+
     configureFlags = [
       "--with-boot-jdk=${openjdk-bootstrap.home}"
       "--with-version-pre="

--- a/pkgs/development/compilers/openjdk/15.nix
+++ b/pkgs/development/compilers/openjdk/15.nix
@@ -59,6 +59,11 @@ let
       patchShebangs --build configure
     '';
 
+    # JDK's build system attempts to specifically detect
+    # and special-case WSL, and we don't want it to do that,
+    # so pass the correct platform names explicitly
+    configurePlatforms = ["build" "host"];
+
     configureFlags = [
       "--with-boot-jdk=${openjdk-bootstrap.home}"
       "--with-version-pre="

--- a/pkgs/development/compilers/openjdk/16.nix
+++ b/pkgs/development/compilers/openjdk/16.nix
@@ -61,6 +61,11 @@ let
       patchShebangs --build configure
     '';
 
+    # JDK's build system attempts to specifically detect
+    # and special-case WSL, and we don't want it to do that,
+    # so pass the correct platform names explicitly
+    configurePlatforms = ["build" "host"];
+
     configureFlags = [
       "--with-boot-jdk=${openjdk-bootstrap.home}"
       "--with-version-build=${version.build}"

--- a/pkgs/development/compilers/openjdk/17.nix
+++ b/pkgs/development/compilers/openjdk/17.nix
@@ -78,6 +78,11 @@ let
       patchShebangs --build configure
     '';
 
+    # JDK's build system attempts to specifically detect
+    # and special-case WSL, and we don't want it to do that,
+    # so pass the correct platform names explicitly
+    configurePlatforms = ["build" "host"];
+
     configureFlags = [
       "--with-boot-jdk=${openjdk-bootstrap.home}"
       "--with-version-build=${version.build}"

--- a/pkgs/development/compilers/openjdk/18.nix
+++ b/pkgs/development/compilers/openjdk/18.nix
@@ -68,6 +68,11 @@ let
       patchShebangs --build configure
     '';
 
+    # JDK's build system attempts to specifically detect
+    # and special-case WSL, and we don't want it to do that,
+    # so pass the correct platform names explicitly
+    configurePlatforms = ["build" "host"];
+
     configureFlags = [
       "--with-boot-jdk=${openjdk-bootstrap.home}"
       "--with-version-build=${version.build}"

--- a/pkgs/development/compilers/openjdk/19.nix
+++ b/pkgs/development/compilers/openjdk/19.nix
@@ -80,6 +80,11 @@ let
       patchShebangs --build configure
     '';
 
+    # JDK's build system attempts to specifically detect
+    # and special-case WSL, and we don't want it to do that,
+    # so pass the correct platform names explicitly
+    configurePlatforms = ["build" "host"];
+
     configureFlags = [
       "--with-boot-jdk=${openjdk-bootstrap.home}"
       "--with-version-build=${version.build}"

